spec/clustering_spec.cr

```diff
@@ -196,4 +196,71 @@ describe LavinMQ::Clustering::Client do
       fail("election campaign did not finish in time, leadership not released on launcher stop?")
     end
   end
+
+  it "wont deadlock under high load when a follower disconnects [#926]" do
+    LavinMQ::Config.instance.clustering_max_unsynced_actions = 1
+    replicator = LavinMQ::Clustering::Server.new(LavinMQ::Config.instance, LavinMQ::Etcd.new("localhost:12379"), 0)
+    tcp_server = TCPServer.new("localhost", 0)
+    spawn(replicator.listen(tcp_server), name: "repli server spec")
+
+    client_io = TCPSocket.new("localhost", tcp_server.local_address.port)
+    client_io.write LavinMQ::Clustering::Start
+    client_io.write_bytes replicator.password.bytesize.to_u8, IO::ByteFormat::LittleEndian
+    client_io.write replicator.password.to_slice
+    client_io.read_byte
+    client_io.write_bytes 2i32, IO::ByteFormat::LittleEndian
```

Comment:
Maybe add some more comments to the spec to make it clearer what is happening and easier to understand in the future? For example why are we writing `2_i32` to `client_io` here and `0i32` later in the spec.